src/lib/customNav/interface.ts

```
@@ -0,0 +1,19 @@
+
+
+export interface QueryParams {
+    [key: string]: string;
+}
+
+// 여러 함수에서 사용 가능한 타입 별칭 정의
+export type QueryFactory = (queryParams?: QueryParams) => Query;
+
+export interface Query {
+    queryKey: (string | QueryParams)[];
+    queryFn: QueryFunction<any, (string | QueryParams)[]>;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)